src/Workspaces/Core/Portable/Utilities/ArraySlice.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Utilities
+{
+    internal struct ArraySlice<T>
+    {
+        private readonly T[] _array;
+        private int _start;
+        private int _length;
+
+        public int Length => _length;
+
+        public ArraySlice(T[] array) : this(array, 0, array.Length)
+        {
+        }
+
+        public ArraySlice(T[] array, TextSpan span) : this(array, span.Start, span.Length)
```

Comment:
Hrmm.  i don't see why.  It's just a generalized "array + range" constructor.   It could work on arrays of any type?  Or are you asking because "TextSpan" implies 'string/chars'?  If so, i think that's not actually the case.  We use that type simply as our own "Range" type.  I'd use "Span", but that's an Editor type, and not available at the Workspace level.  So TextSpan is our equivalent.